Plugin/OrderRepositoryPlugin.php

```diff
@@ -20,11 +21,26 @@ public function __construct(OrderPaymentExtensionFactory $orderPaymentExtensionF
      * and add the additional information as a json encoded string to an extension attribute.
      */
     public function afterGet(OrderRepositoryInterface $orderRepository, OrderInterface $order)
+    {
+        $this->changeAdditionalInformation($order);
+        return $order;
+    }
+
+    public function afterGetList(OrderRepositoryInterface $orderRepository, OrderSearchResultInterface $orders)
+    {
+        foreach ($orders->getItems() as $key => $order)
+        {
+            $this->changeAdditionalInformation($order);
+        }
+
+        return $orders;
+    }
+
+    private function changeAdditionalInformation(OrderInterface &$order): OrderInterface
```

Comment:
we should avoid calling stuff by reference (changing and returning an `OrderInterface` item will do finely)

Reply:
Even if its in a set?